src/middlewared/middlewared/plugins/auth.py

```diff
@@ -161,13 +160,14 @@ def dump(self):
 
 
 def is_internal_session(session):
-    if isinstance(session.app.origin, UnixSocketOrigin) and session.app.origin.uid == 0:
-        return True
-
-    if isinstance(session.app.authenticated_credentials, TrueNasNodeSessionManagerCredentials):
-        return True
-
-    return False
+    return any((
+        all((
```

Comment:
Why `any` and `all`?

Reply:
the first `all` requires 2 conditionals to be True, which is the first boolean in the tuple to `any`. It's the same as:
```
if (true and true) or (true)

Reply:
`all((x, y))` will evaluate `y` even if `x` is false while `x and y` will only evaluate `x`. Similarly, `any((x, y))` will evaluate `y` even if `x` is true, while `x or y` will only evaluate `x`. These differences can lead to bugs, e.g. https://github.com/truenas/middleware/pull/14281

Reply:
Very well, I'll fix

Reply:
fixed